src/storage-utils/addresses.ts

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import AddressMetadataStorage from '@/persistent-storage/address-metadata'
+import { DataKey } from '@/persistent-storage/encrypted-storage'
+import { newAddressesGenerated, syncAddressesData } from '@/store/addressesSlice'
+import { store } from '@/store/store'
+import { AddressBase } from '@/types/addresses'
+
+export const saveNewAddresses = (addresses: AddressBase[], dataKey: DataKey) => {
+  addresses.forEach((address) =>
+    AddressMetadataStorage.store({
+      dataKey,
+      index: address.index,
+      settings: {
+        isDefault: address.isDefault,
+        label: address.label,
+        color: address.color
+      }
+    })
+  )
+
+  store.dispatch(newAddressesGenerated(addresses))
+  store.dispatch(syncAddressesData(addresses.map((address) => address.hash)))
+}
```

Comment:
@mvaivre I am curious to hear your thoughts on this new folder called `storage-utils`. My goal with this is to have a place where utility functions that affect both the local storage as well as the Redux store live. It didn't feel good to put them inside `@/persistant-storage`, nor in `@/store` because I didn't want to create deps between them. So I created `@/storage-utils`. I am not 100% convinced with the name. Other names I thought was `@/storage` but that's confusing, next to `@/store` and `@/persistant-storage`.

A second idea would be to restructure the `@/store` directory and instead of just adding `xyzSlice.ts` files, to have a folder structure like so:

<img width="546" alt="image" src="https://user-images.githubusercontent.com/1579899/217490360-dc7a98f6-7dd4-4b21-9dd5-b1de410d329e.png">

Reply:
Thanks for the heads-up. I saw this one yesterday and was not sure about how I felt about it. Today it's a bit clearer :)

- I agree with you on the first point. Those functions should be decoupled from both `store` and `persistent-storage`. They should be one level above in the tree IMO.
- The second idea looks good on paper, but only if those utils functions were used exclusively in the sibling slice found next to them in the tree. It's a bit weird IMO to bury utils functions like `saveNewWallet` in the hierarchy, when they are used globally. 

My proposition would be something like :

```
storage
├── app-state
│   ├── store.ts
│   ├── actions.ts
│   └── slices
│       ├── addressesSlice.ts
│       ├── networkSlice.ts
│       └── ...
├── persistant-store
│   ├── addressesPersitance.ts
│   ├── contactsPersistance.ts
│   └── ...
└── storage-utils
    ├── addressesStorageUtils.ts
    ├── walletStorageUtils.ts
    └── ...
```

Note that I used longer but more explicit file names to help with IDE search :)

Reply:
I think your proposal is awesome.

Reply:
I'll go for this!

<img width="525" alt="image" src="https://user-images.githubusercontent.com/1579899/217757396-b0ca4e2f-6fc5-4d0e-b8f5-bc4307a867b5.png">